lib/compressMeshes.js

```diff
@@ -0,0 +1,524 @@
+}
+
+function compressMeshes(gltf, options) {
+    addExtensionsRequired(gltf, 'KHR_draco_mesh_compression');
+    var hashPrimitives = [];
+    options = options === undefined ? {} : options;
+    var compressionLevel = !defined(options.compressionLevel) ? 7 : options.compressionLevel;
+    var positionQuantization = !defined(options.quantizePosition) ? 14 : options.quantizePosition;
+    var normalQuantization = options.quantizeNormal === undefined ? 10 : options.quantizeNormal;
+    var texcoordQuantization = options.quantizeTexcoord === undefined ? 12 : options.quantizeTexcoord;
+    var colorQuantization = options.quantizeColor === undefined ? 8 : options.quantizeColor;
+    var skinQuantization = options.quantizeSkin === undefined ? 12 : options.quantizeSkin;
+    var useUnifiedQuantization = options.unifiedQuantization === undefined ? false : options.unifiedQuantization;
+    var positionOrigin, positionRange;
+
+    if (useUnifiedQuantization) {
+      // Collect bounding box from all primitives. Currently works only for vec3 positions (XYZ).
+      var accessors = gltf.accessors;
+      var min = arrayFill(new Array(3), Number.POSITIVE_INFINITY);
+      var max = arrayFill(new Array(3), Number.NEGATIVE_INFINITY);
+      ForEach.accessorWithSemantic(gltf, "POSITION", function(accessorId, attributeSemantic, primitive) {
+          var accessor = accessors[accessorId];
+          if (accessor.type !== 'VEC3') {
+            console.log("Couldn't perform unified quantization. Input contains position accessor with an unsupported number of components.");
+            useUnifiedQuantization = false;
+            return;
+          }
+          var accessorMin = accessor.min;
+          var accessorMax = accessor.max;
+          for (var j = 0; j < 3; ++j) {
+            min[j] = Math.min(min[j], accessorMin[j]);
+            max[j] = Math.max(max[j], accessorMax[j]);
+          }
+      });
+      positionOrigin = min;
+      positionRange = Math.max(max[0] - min[0], Math.max(max[1] - min[1], max[2] - min[2]));
+    }
+
+    ForEach.mesh(gltf, function(mesh) {
+        ForEach.meshPrimitive(mesh, function(primitive, primitiveId) {
+            var primitiveType = primitive.mode;
+            // Only support triangles now.
+            if (defined(primitive.mode) && primitive.mode !== 4) {
+              console.log("Skipped primitive. Unsupported primitive mode.");
+              return;
+            }
+
+            // TODO: Use hash map.
+            var primitiveGeometry = {
+              attributes : primitive.attributes,
+              indices : primitive.indices,
+              mode : primitive.mode
+            };
+            var hashValue = hashObject(primitiveGeometry);
+            if (hashPrimitives[hashValue] !== undefined) {
+              console.log('Found duplicate!');
+              // Copy compressed primitive.
+              copyCompressedExtensionToPrimitive(
+                  primitive, hashPrimitives[hashValue]);
```

Comment:
Formatting: Combine into same line as the function.

Reply:
Done.